### ISSUE TITLE
Unexpected `E0277` with Procedural Derive Macro

### ISSUE THREAD

**Issue by HTG-YT** [intellij-rust/intellij-rust#9975]
## Environment

* **IntelliJ Rust plugin version:** 0.4.187.5137-223-nightly
* **Rust toolchain version:** rustc 1.68.0-nightly (0b90256ad 2023-01-13)
* **IDE name and version:** IntelliJ IDEA 2022.3.2 Preview (Ultimate Edition) Build #IU-223.8617.9, built on January 11, 2023
* **Operating system:** Windows 11 22H2 (Build 22621.1105)

## Problem description
Consider the following traits:
```rs
pub trait Command: CommandMetadata {}
pub trait CommandMetadata {}
```
And `CommandMetadata` having a procedural derive macro for deriving it.

When a struct has the attribute `#[derive(CommandMetadata)]` on it, the struct should automatically satisfy the `: CommandMetadata` bound on the `Command trait`.

However, the plugin highlights the `impl Command for` declaration, showing that the struct does not satisfy the `: CommandMetadata` bound.

## Steps to reproduce

1. Clone this repository: https://github.com/HTG-YT/HarTex
2. Navigate to the `discord-frontend/hartex-discord-commands/src/general/ping.rs`
3. Observe the unexpected error:
![image](https://user-images.githubusercontent.com/39023054/212464680-65d16473-c812-4fd3-abe9-f50ba7670c1c.png)

**Comment by Undin** [comment]
The plugin can't expand `derive(CommandMetadata)` proc macro call (for me it expands to `compile_error! {"expected identifier; found `true`" }`). As a result, the plugin doesn't know about `CommandMetadata` implementation generated by this macro and reports an error.

@vlad20012 Could you check it, please?

**Comment by HTG-YT** [comment]
There have been some updates to the macro, so perhaps you may test again pulling this commit: https://github.com/TeamHarTex/HarTex/commit/41126ed743bc8b574a54d1a0b12153bcd6321342
(Also, there are no compiler errors whatsoever when compiling the `discord-frontend` workspace.)

**Comment by HTG-YT** [comment]
Here is the output from `cargo expand` for your reference:
```rust
    use hartex_discord_commands_core::traits::Command;
    use hartex_discord_commands_macros::CommandMetadata;
    use hartex_discord_core::discord::model::application::interaction::Interaction;
    #[metadata(command_type = 1)]
    #[metadata(interaction_only = true)]
    #[metadata(name = "ping")]
    pub struct Ping;
    extern crate hartex_discord_commands_core as _commands_core;
    #[automatically_derived]
    impl _commands_core::traits::CommandMetadata for Ping {
        fn command_type(&self) -> u8 {
            1
        }
        fn interaction_only(&self) -> bool {
            true
        }
        fn name(&self) -> String {
            String::from("ping")
        }
    }
    impl Command for Ping {
        #[allow(clippy::unused_async)]
        async fn execute(&self, _: Interaction) -> hartex_discord_eyre::Result<()> {
            ::core::panicking::panic("not yet implemented")
        }
    }
```
